tests-integ/test_model_bedrock.py

```diff
@@ -0,0 +1,116 @@
+import pytest
+
+import strands
+from strands import Agent
+from strands.models import BedrockModel
+
+
+@pytest.fixture
+def system_prompt():
+    return "You are an AI assistant that uses & instead of ."
+
+
+@pytest.fixture
+def streaming_model():
+    return BedrockModel(
+        model_id="us.anthropic.claude-3-7-sonnet-20250219-v1:0",
+        streaming=True,
+    )
+
+
+@pytest.fixture
+def non_streaming_model():
+    return BedrockModel(
+        model_id="us.meta.llama3-2-90b-instruct-v1:0",
+        streaming=False,
+    )
+
+
+@pytest.fixture
+def streaming_agent(streaming_model, system_prompt):
+    return Agent(model=streaming_model, system_prompt=system_prompt)
+
+
+@pytest.fixture
+def non_streaming_agent(non_streaming_model, system_prompt):
+    return Agent(model=non_streaming_model, system_prompt=system_prompt)
+
+
+def test_streaming_agent(streaming_agent):
+    """Test agent with streaming model."""
+    result = streaming_agent("Hello!")
+
+    assert len(str(result)) > 0
+
+
+def test_non_streaming_agent(non_streaming_agent):
+    """Test agent with non-streaming model."""
+    result = non_streaming_agent("Hello!")
+
+    assert len(str(result)) > 0
+
+
+def test_streaming_model_events(streaming_model):
+    """Test streaming model events."""
+    messages = [{"role": "user", "content": [{"text": "Hello"}]}]
+
+    # Call converse and collect events
+    events = list(streaming_model.converse(messages))
+
+    # Verify basic structure of events
+    assert any("messageStart" in event for event in events)
+    assert any("contentBlockDelta" in event for event in events)
+    assert any("messageStop" in event for event in events)
+
+
+def test_non_streaming_model_events(non_streaming_model):
+    """Test non-streaming model events."""
+    messages = [{"role": "user", "content": [{"text": "Hello"}]}]
+
+    # Call converse and collect events
+    events = list(non_streaming_model.converse(messages))
+
+    # Verify basic structure of events
+    assert any("messageStart" in event for event in events)
+    assert any("contentBlockDelta" in event for event in events)
+    assert any("messageStop" in event for event in events)
+
+
+def test_tool_use_streaming(streaming_model):
+    """Test tool use with streaming model."""
+
+    @strands.tool
+    def calculator(expression: str) -> float:
+        """Calculate the result of a mathematical expression."""
+        return eval(expression)
+
+    agent = Agent(model=streaming_model, tools=[calculator])
+    result = agent("What is 123 + 456?")
+
+    # Print the full message content for debugging
+    print("\nFull message content:")
+    import json
+
+    print(json.dumps(result.message["content"], indent=2))
+
+    # The test is passing as long as the agent successfully uses the tool
```

Comment:
> The test is passing as long as the agent successfully uses the tool

We can assert that the tool was called by having the tool call set a variable, no?

```
def calculator():
   nonlocal tool_was_called
   tool_was_called = True
```

Reply:
Same with below.